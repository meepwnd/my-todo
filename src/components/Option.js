import React from "react";
import { connect } from "react-redux";
import { removeOption } from "../actions/actions";
import { toggleCompletion } from "../actions/actions";

export const Option = props => (
  <li className="option">
    <span
      onClick={() => {
        props.toggleCompletion(props.id);
      }}
      style={{ textDecoration: props.completed ? "line-through" : "none" }}
    >
      {props.option}
    </span>

    <button
      className="btn btn-remove"
      onClick={() => {
        props.removeOption(props.id);
      }}
    >
      Remove
    </button>
  </li>
);

const mapDispatchToProps = dispatch => ({
  removeOption: id => dispatch(removeOption(id)),
  toggleCompletion: id => dispatch(toggleCompletion(id))
});

export default connect(
  null,
  mapDispatchToProps
)(Option);
