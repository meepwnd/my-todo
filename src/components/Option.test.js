import React from "react";
import { mount } from "enzyme";
import { Option } from "./Option";

describe("Option component", () => {
  const toggleCompletion = jest.fn();
  const removeOption = jest.fn();

  const props = {
    toggleCompletion,
    removeOption
  };

  const wrapper = mount(<Option {...props} />);

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call toggleCompletion function on click", () => {
    wrapper.find("span").simulate("click");
    expect(toggleCompletion).toHaveBeenCalled();
  });

  it("should call removeOption function on button click", () => {
    wrapper.find("button").simulate("click");
    expect(removeOption).toHaveBeenCalled();
  });
});
