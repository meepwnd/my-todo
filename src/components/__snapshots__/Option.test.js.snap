// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option component should render correctly 1`] = `
<Option
  removeOption={[MockFunction]}
  toggleCompletion={[MockFunction]}
>
  <li
    className="option"
  >
    <span
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    />
    <button
      className="btn btn-remove"
      onClick={[Function]}
    >
      Remove
    </button>
  </li>
</Option>
`;
